src/autopas/selectors/tuningStrategy/PredictiveTuning.h

```diff
@@ -0,0 +1,370 @@
+                        allowedDataLayoutOptions, allowedNewton3Options);
+  }
+
+  /**
+   * Constructor for the PredictiveTuning that only contains the given configurations.
+   * This constructor assumes only valid configurations are passed! Mainly for easier unit testing.
+   * @param allowedConfigurations Set of configurations AutoPas can choose from.
+   */
+  explicit PredictiveTuning(std::set<Configuration> allowedConfigurations)
+      : _containerOptions{}, _searchSpace(std::move(allowedConfigurations)), _currentConfig(_searchSpace.begin()) {
+    for (auto config : _searchSpace) {
+      _containerOptions.insert(config.container);
+    }
+  }
+
+  inline const Configuration &getCurrentConfiguration() const override { return *_currentConfig; }
+
+  inline void removeN3Option(Newton3Option badNewton3Option) override;
+
+  inline void addEvidence(long time) override {
+    _traversalTimes[*_currentConfig] = time;
+    _traversalTimesStorage[*_currentConfig].emplace_back(std::make_pair(_timer, time));
+  }
+
+  inline void reset() override {
+    _traversalTimes.clear();
+    _traversalPredictions.clear();
+    _optimalSearchSpace.clear();
+    // Too expensive?
+    _validSearchSpace = _searchSpace;
+    _validConfiguration = false;
+    selectPredictedConfigurations();
+
+    // sanity check
+    if (_optimalSearchSpace.empty()) {
+      autopas::utils::ExceptionHandler::exception("PredicitveTuning: No possible configuration prediction found!");
+    }
+
+    _currentConfig = _searchSpace.begin();
+    while (_optimalSearchSpace.count(*_currentConfig) == 0 && _currentConfig != _searchSpace.end()) {
+      ++_currentConfig;
+    }
+  }
+
+  inline bool tune(bool = false) override;
+
+  inline std::set<ContainerOption> getAllowedContainerOptions() const override { return _containerOptions; }
+
+  inline bool searchSpaceIsTrivial() const override { return _searchSpace.size() == 1; }
+
+  inline bool searchSpaceIsEmpty() const override { return _searchSpace.empty(); }
+
+ private:
+  /**
+   * Fills the search space with the cartesian product of the given options (minus invalid combinations).
+   * @param allowedContainerOptions
+   * @param allowedTraversalOptions
+   * @param allowedDataLayoutOptions
+   * @param allowedNewton3Options
+   */
+  inline void populateSearchSpace(const std::set<ContainerOption> &allowedContainerOptions,
+                                  const std::set<double> &allowedCellSizeFactors,
+                                  const std::set<TraversalOption> &allowedTraversalOptions,
+                                  const std::set<DataLayoutOption> &allowedDataLayoutOptions,
+                                  const std::set<Newton3Option> &allowedNewton3Options);
+
+  inline void selectOptimalConfiguration();
+
+  /**
+   * Selects the configuration that are going to be tested.
+   */
+  inline void selectPredictedConfigurations();
```

Comment:
```suggestion
  /**
   * Selects the configurations that are going to be tested.
   */
  inline void selectOptimalSearchSpace();
```
(don't apply this suggestion through github but make the refactoring manually)